rfcs/text/design-tokens.md

```diff
@@ -0,0 +1,154 @@
+
+This approach uses themes feature from `styled-components` since it is already implemented and uses javascript
+variables.
+
+At the moment, we have two levels of design tokens:
+
+- **Generic tokens** - kind of palette with all the values and scales.
+- **Theme tokens** - tokens for a theme, which contain references to generic tokens.
+
+So themes can be for example: *S Biz*, *M Biz*, *High Contrast*, *Dark Mode* and so on.
+
+Themes can overwrite not only colors, but also sizes. This way it is possible to create size themes, such as *Small*
+or *Medium* or *Large*
+
+In the end it all depends on designers, how they will build components in Figma, and which tokens they will use.
+
+# Drawbacks
+
+In case of CSS approach adaptation, there would be two theming systems available for carbon for the time of
+implementation.
+
+# Alternatives
+
+## Design tokens via styled-components and CSS variables
+
+Design token library provides also tokens in scss and css variables formats, therefore it is possible to implement
+theming using those technologies. Using styled-components in combination with css variables would look like this:
+
+```js
+const Container = styled.div`
+  background: var(--sageds-colors-base-400);
+  color: var(--sageds-colors-text-white);
+`;
+```
+
+Then, switching themes would be handled by including or excluding given stylesheets.
+
+## Tag function for generating token
+
+Javascript tag function can be created. It could take string describing token, and depending on its implmentation, it
+could transform it to the `var(--token)` string, or return value of the token
+
+```js
+function token(token) {
+  return designTokens[_.camelCase(`sageds.${token[0]}`)];
+}
+```
+
+OR
+
+```js
+function token(token) {
+  return `var(--sageds-${token[0].split('.').join('-')})`;
+}
+```
+
+usage:
+
+```js
+const Container = styled.div`
+  background: ${token`colors.base.400`};
+  color: ${token`colors.text.white`};
+`;
+```
+
+# Adoption strategy
+
+Design tokens will be available globally in carbon. This means, that developers will be able to choose design tokens
+over current theme variables. However, the main process of design system alignment (which includes design tokens
+migration) will be handled by Design System Development Team and this will be the team, that will be most interested in
+design tokens.
+
+While aligning component, only design tokens should be used. This implies, that design tokens library should be updated
+with the tokens that are needed for aligning given component. If there are any tokens missing, then it needs to be
+reported to Design System Team and situation needs to be explained.
+
+Whichever approach (JS or CSS) would be chosen, the adoption will consist in rewriting current styles of components
+using newly provided tokens (JS approach), or just entering CSS variables (CSS approach).
+
+In specific, justified cases, developer can use generic tokens. However, generic tokens shouldn't be referenced
```

Comment:
Generic tokens would contain wider range of predefined colors, and other tokens, for example: 
```js
const tokens = {
  colorsSlate50:  "#f2f5f6",
  colorsSlate100: "#e6ebed",
  colorsSlate200: "#ccd6db",
  colorsSlate300: "#b3c2c8",
  colorsSlate500: "#8099a4",
  colorsSlate600: "#668592",
  colorsSlate700: "#4d7080",
  colorsSlate800: "#335c6d",
  colorsSlate900: "#19475b",

  colorsAegean100: "#4DA0D9",
  colorsAegean200: "#1985CE",
  colorsAegean300: "#1985CE",
  colorsAegean400: "#0073C2",
  colorsAegean500: "#006BB4",
  colorsAegean600: "#005C9A",
  colorsAegean700: "#00538C",
  colorsAegean800: "#004472",
  colorsAegean900: "#003C64",

  spacing100: "2px",
  spacing200: "8px",
  spacing300: "8px",
  spacing400: "16px",
  spacing500: "32px",
  spacing600: "64px",
  spacing700: "72px",
  spacing800: "80px",
  spacing900: "88px",

  fontFamiliesBody: "Lato",
  fontFamiliesTitle: "Roboto"

  [...]
},
```

Reply:
However on the last meeting were thinking of not including this one in design tokens library, since according to design sytem, only predefined colors should be used.

Reply:
How does this differ from the base theme, wouldn't all colors be defined there?

Reply:
Using deisgn tokens will also require kind of base theme, exactly the same way as it is done now.